src/commands/typescript/update.ts

```diff
@@ -54,7 +59,41 @@ export default class Update extends SfdxCommand {
     }
   }
 
+  private isLernaRepo(): boolean {
+    const lernaConfigPath = path.resolve('lerna.json');
+    return fs.existsSync(lernaConfigPath);
+  }
+
+  private async getPackagePaths(): Promise<string[]> {
+    const workingDir = pwd().stdout;
+    const lernaJson = (await fs.readJson('lerna.json')) as LernaJson;
+    const packageGlobs = lernaJson.packages || ['*'];
+    const packages = packageGlobs
+      .map((pGlob) => glob.sync(pGlob))
+      .reduce((x, y) => x.concat(y), [])
+      .map((pkg) => path.join(workingDir, pkg));
+    return packages;
+  }
+
   private async updateEsTarget(): Promise<void> {
+    if (this.isLernaRepo()) {
+      const packagePaths = await this.getPackagePaths();
+
+      for (let i: number; i < packagePaths.length; i++) {
+        const packagePath = packagePaths[i];
+        const tsConfigPath = path.join(packagePath, 'tsconfig.json');
+        const tsConfigString = await fs.readFile(tsConfigPath, 'utf-8');
+
+        // strip out any comments that might be in the tsconfig.json
+        const commentRegex = new RegExp(/(\/\/.*)/, 'gi');
+        const tsConfig = JSON.parse(tsConfigString.replace(commentRegex, '')) as AnyJson;
+
+        set(asObject(tsConfig), 'compilerOptions.target', this.flags.target);
+        this.ux.log(`Updating tsconfig target at ${tsConfigPath} to:`, this.flags.target);
+        await fs.writeJson(tsConfigPath, tsConfig);
+      }
+    }
+
     const tsConfigPath = path.resolve('tsconfig.json');
     const tsConfigString = await fs.readFile(tsConfigPath, 'utf-8');
```

Comment:
Not all lerna repos will have a root-level tsconfig, but I think they can. Maybe wrap this and the following in a try-catch and only throw if we aren't in a lerna project? That's assuming this should throw like it does today?

Reply:
There is this fx that takes care of opening `tsconfig.json` files from the `packages` directory in a lerna repo: 
![image](https://user-images.githubusercontent.com/2845947/112696323-5e9aa900-8e64-11eb-9b59-fa1487ba381a.png)
Regular repos just read their `tsconfig.json` from the root of the repo like they did before